### PR TITLE
NO-JIRA:  Adding lint configuration

### PR DESCRIPTION
**Pull request by carnellj-genesys**

### CODE REVIEW

File: golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+linters:
```
Comment (carnellj-genesys) [inline golangci.yaml:1]: Added the linting configuration  for the tool and the ran it against one package to try it out.

File: genesyscloud/architect_datatable/resource_genesyscloud_architect_datatable_proxy.go [hunk 2]
```diff
@@ -170,9 +168,7 @@ func getAllArchitectDatatableFn(ctx context.Context, p *architectDatatableProxy)
 			break
 		}
 
-		for _, table := range *tables.Entities {
-			totalRecords = append(totalRecords, table)
-		}
+		totalRecords = append(totalRecords, *tables.Entities...)
```
Comment (carnellj-genesys) [inline genesyscloud/architect_datatable/resource_genesyscloud_architect_datatable_proxy.go:171]: Simplified the code for appending entities

File: genesyscloud/architect_datatable/resource_genesyscloud_architect_datatable_proxy.go [hunk 3]
```diff
@@ -74,7 +74,7 @@ func createOrUpdateArchitectDatatableFn(ctx context.Context, p *architectDatatab
 	// create path and map variables
 	path := p.architectApi.Configuration.BasePath + "/api/v2/flows/datatables"
 
-	if createAction == false {
+	if !createAction {
```
Comment (carnellj-genesys) [inline genesyscloud/architect_datatable/resource_genesyscloud_architect_datatable_proxy.go:77]: Fixed per comments from Linter

File: genesyscloud/architect_datatable/resource_genesyscloud_architect_datatable.go [hunk 4]
```diff
@@ -115,19 +115,19 @@ func readArchitectDatatable(ctx context.Context, d *schema.ResourceData, meta in
...
 		}
 		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceArchitectDatatable())
-		d.Set("name", *datatable.Name)
-		d.Set("division_id", *datatable.Division.Id)
+		_ = d.Set("name", *datatable.Name)
+		_ = d.Set("division_id", *datatable.Division.Id)
 
 		if datatable.Description != nil {
-			d.Set("description", *datatable.Description)
+			_ = d.Set("description", *datatable.Description)
```
Comment (carnellj-genesys) [inline genesyscloud/architect_datatable/resource_genesyscloud_architect_datatable.go:122]: Explicitly showed I was not going to use the error code. This makes it easier for devs to realize an error code does get returning, but we are specifically not doing anything with it.

File: genesyscloud/architect_datatable/genesyscloud_architect_datatable_init_test.go [hunk 5]
```diff
@@ -20,11 +20,15 @@ type registerTestInstance struct {
 
 // registerTestResources registers all resources used in the tests
 func (r *registerTestInstance) registerTestResources() {
+	r.resourceMapMutex.Lock()
```
Comment (carnellj-genesys) [inline genesyscloud/architect_datatable/genesyscloud_architect_datatable_init_test.go:23]: These were missed during the code review.  We should be locking the map so we avoid any potential concurrent map issues,

**Review by dginty4 — APPROVED** [review]

File: genesyscloud/architect_datatable/genesyscloud_architect_datatable_init_test.go (hunk 5 above)
Comment (dginty4) [reply genesyscloud/architect_datatable/genesyscloud_architect_datatable_init_test.go:23]: Was it the linter that caught this?

Comment (carnellj-genesys) [reply genesyscloud/architect_datatable/genesyscloud_architect_datatable_init_test.go:23]: It caught that the variables were unused.   Once I knew they were unused, I realized that we need to include the locking in there.

**Pull request merged**